test/tools/d_do_test.d

```diff
@@ -596,24 +600,52 @@ bool compareOutput(string output, string refoutput, const ref EnvData envData)
             import std.ascii : isDigit;
             output.skipOver!isDigit();
         }
-        else // $p:<some path>$
+        else // $<identifier>:<special content>$
         {
-            // Find the end of the expected path
-            /// ( expected path tail, "$", remaining expected output )
+            /// ( special content, "$", remaining expected output )
             auto refparts = refoutput.findSplit("$");
-            enforce(refparts, "Malformed path sequence!");
+            enforce(refparts, "Malformed special sequence!");
+            refoutput = refparts[2];
 
-            // Substitute / with the appropriate directory separator
-            auto pathEnd = refparts[0].replace("/", envData.sep);
+            if (special.id == 2) // $p:<some path>$
+            {
+                // special content is the expected path tail
+                // Substitute / with the appropriate directory separator
+                auto pathEnd = refparts[0].replace("/", envData.sep);
 
-            /// ( whole path, remaining output )
-            auto parts = output.findSplitAfter(pathEnd);
+                /// ( whole path, remaining output )
+                auto parts = output.findSplitAfter(pathEnd);
 
-            if (!parts || !exists(parts[0]))
-                return false;
+                if (!parts || !exists(parts[0]))
+                    return false;
 
-            refoutput = refparts[2];
-            output = parts[1];
+                output = parts[1];
```

Comment:
Ditto with a `continue` here.

Reply:
I'd rather not do this because I might add another special token of this kind in the future.